plutus-tx-plugin/src/PlutusTx/Options.hs

```diff
@@ -0,0 +1,234 @@
+           in (k, PluginOption typeRep (setFalse k) poDoSimplifierInline showText mempty)
+        , let k = "no-simplifier-remove-dead-bindings"
+           in (k, PluginOption typeRep (setFalse k) poDoSimplifierRemoveDeadBindings showText mempty)
+        , let k = "profile-all"
+           in (k, PluginOption typeRep (flag (const All) k) poProfile showText mempty)
+        , let k = "coverage-all"
+           in (k, PluginOption typeRep (setTrue k) poCoverageAll showText mempty)
+        , let k = "coverage-location"
+           in (k, PluginOption typeRep (setTrue k) poCoverageLocation showText mempty)
+        , let k = "coverage-boolean"
+           in (k, PluginOption typeRep (setTrue k) poCoverageBoolean showText mempty)
+        , let k = "remove-trace"
+           in (k, PluginOption typeRep (setTrue k) poRemoveTrace showText mempty)
+        ]
+  where
+    showText :: Show a => a -> Text
+    showText = Text.pack . show
+
+flag :: (a -> a) -> OptionKey -> Maybe OptionValue -> Validation ParseError (a -> a)
+flag f k = maybe (Success f) (Failure . UnexpectedValue k)
+
+setTrue :: OptionKey -> Maybe OptionValue -> Validation ParseError (Bool -> Bool)
+setTrue = flag (const True)
+
+setFalse :: OptionKey -> Maybe OptionValue -> Validation ParseError (Bool -> Bool)
+setFalse = flag (const False)
+
+intOption :: OptionKey -> Maybe OptionValue -> Validation ParseError (Int -> Int)
+intOption k = \case
+    Just v ->
+        maybe
+            (Failure (CannotParseValue k v (someTypeRep (Proxy @Int))))
+            (Success . const)
+            (readMaybe (Text.unpack v))
+    Nothing -> Failure (MissingValue k)
+
+defaultPluginOptions :: PluginOptions
+defaultPluginOptions =
+    PluginOptions
+        { _poDoTypecheck = True
+        , _poDeferErrors = False
+        , _poContextLevel = 1
+        , _poDumpPir = False
+        , _poDumpPlc = False
+        , _poDumpUPlc = False
+        , _poOptimize = True
+        , _poPedantic = False
+        , _poVerbose = False
+        , _poDebug = False
+        , _poMaxSimplifierIterations = view PIR.coMaxSimplifierIterations PIR.defaultCompilationOpts
+        , _poDoSimplifierUnwrapCancel = True
+        , _poDoSimplifierBeta = True
+        , _poDoSimplifierInline = True
+        , _poDoSimplifierRemoveDeadBindings = True
+        , _poProfile = None
+        , _poCoverageAll = False
+        , _poCoverageLocation = False
+        , _poCoverageBoolean = False
+        , _poRemoveTrace = False
+        }
+
+processOne ::
+    OptionKey ->
+    Maybe OptionValue ->
+    Validation ParseError (PluginOptions -> PluginOptions)
+processOne key val = case Map.lookup key pluginOptions of
+    Just (PluginOption _ f field _ _) -> over field <$> f val
+    Nothing ->
+        let suggs =
+                Text.pack
+                    <$> GHC.fuzzyMatch (Text.unpack key) (Text.unpack <$> Map.keys pluginOptions)
+         in Failure (UnrecognisedOption key suggs)
+
+processAll ::
```

Comment:
Personally, I find a bunch of these functions to be a bit too point-free to be easy to read.